NuRadioReco/modules/triggerTimeAdjuster.py

```diff
@@ -97,68 +100,74 @@ def run(self, event, station, detector, mode='sim_to_data'):
                             cut_samples_beginning = trigger_time_sample - samples_before_trigger
                             roll_by = 0
-                            if(cut_samples_beginning + number_of_samples > trace_length):
-                                logger.info("trigger time is sample {} but total trace length is only {} samples (requested trace length is {} with an offest of {} before trigger). To achieve desired configuration, trace will be rolled".format(
-                                    trigger_time_sample, trace_length, number_of_samples, samples_before_trigger))
+                            if cut_samples_beginning + number_of_samples > trace_length:
+                                logger.info(f"trigger time is sample {trigger_time_sample} but total "
+                                             f"trace length is only {trace_length} samples "
+                                             f"(requested trace length is {number_of_samples} with an offest of "
+                                             f"{samples_before_trigger} before trigger). "
+                                              "To achieve desired configuration, trace will be rolled")
+
                                 roll_by = cut_samples_beginning + number_of_samples - trace_length  # roll_by is positive
                                 trace = np.roll(trace, -1 * roll_by)
                                 cut_samples_beginning -= roll_by
-                            rel_station_time_samples = cut_samples_beginning + roll_by
-                        elif(samples_before_trigger > trigger_time_sample):
-                            roll_by = -trigger_time_sample + samples_before_trigger
+
+                        elif samples_before_trigger > trigger_time_sample:
+                            roll_by = samples_before_trigger - trigger_time_sample
                             logger.info(
                                 "trigger time is before 'trigger offset window', the trace needs to be rolled by {} samples first".format(roll_by))
+
                             trace = np.roll(trace, roll_by)
-                            trigger_time_sample -= roll_by
-                            rel_station_time_samples = -roll_by
 
                         # shift trace to be in the correct location for cutting
-                        # logger.debug(f"cutting trace to {cut_samples_beginning}-{number_of_samples + cut_samples_beginning} samples")
+                        logger.debug(f"cutting trace to {cut_samples_beginning}-{number_of_samples + cut_samples_beginning} samples")
                         trace = trace[cut_samples_beginning:(number_of_samples + cut_samples_beginning)]
                         channel.set_trace(trace, channel.get_sampling_rate())
-                        channel.set_trace_start_time(trigger_time)
+                        channel.set_trace_start_time(trigger_time - pre_trigger_time)
```

Comment:
that should be the relevant change!

Reply:
At first glance this looks like the right thing to do here, yes
... but it doesn't solve our main issue with the RNO-G sims, where the pulses triggered on may be jumping around in the clipped traces. The `channel.set_trace` is still writing the identical clipped trace